File: src/simulation/culture/aspect/dependency/Dependency.java
package simulation.culture.aspect.dependency;

import simulation.culture.aspect.Aspect;
import simulation.culture.aspect.AspectController;
import simulation.culture.aspect.AspectResult;
import simulation.culture.group.AspectCenter;
import simulation.space.resource.tag.ResourceTag;
import simulation.culture.group.Group;

public interface Dependency {
    String getName();
    boolean isCycleDependency(Aspect aspect);
    boolean isCycleDependencyInner(Aspect aspect);
    AspectResult useDependency(AspectController controller);
    boolean isPhony();
    ResourceTag getType();
    Dependency copy();
    void swapDependencies(AspectCenter aspectCenter);
}

File: src/simulation/culture/aspect/dependency/Dependency.kt
package simulation.culture.aspect.dependency

import simulation.culture.aspect.Aspect
import simulation.culture.aspect.AspectController
import simulation.culture.aspect.AspectResult
import simulation.culture.group.centers.AspectCenter

interface Dependency {
    val name: String
    fun isCycleDependency(otherAspect: Aspect): Boolean
    fun isCycleDependencyInner(otherAspect: Aspect): Boolean
    fun useDependency(controller: AspectController): AspectResult
    val isPhony: Boolean
    fun copy(): Dependency
    fun swapDependencies(aspectCenter: AspectCenter)
}